### PR TITLE
BLD: fix compilation errors with Cython 3.0.0b1 (pow return type)

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

Fix most of the remaining compilation errors with Cython 3.0.0b1

Address the current top blocking problem in #4355
For context see https://github.com/cython/cython/issues/5299

### CODE REVIEW

**Review by chrishavlin — DISMISSED** [review]
Generally looks good -- one naive question: if `cython.cpow` is introduced in 3.0, does 2.9x just ignore it?

**Comment by neutrinoceros** [comment]
Great question, I actually hadn't noticed that this decorator was new in Cython 3.0 and now I'm surprised that Cython 0.29 seems just happy to swallow it. I upstreamed your question (see https://github.com/cython/cython/issues/5299#issuecomment-1464560241)

**Comment by neutrinoceros** [comment]
So, this decorator was added to Cython 0.29.x as a no-op for backward compatibility, see [Cython's changelog](https://cython.readthedocs.io/en/latest/src/changes.html#id34)
but it *does* require Cython>=0.29.33, so I'll need to bump the requirement explicitly.

**Review by chrishavlin — APPROVED** [review]

File: .github/workflows/wheels.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
     - name: Install build time dependencies
       shell: bash
       run: |
-        python -m pip install "Cython>=0.29.21,<3.0"
```
Comment (neutrinoceros) [inline .github/workflows/wheels.yaml:87]: forced-pushed again because I forgot to update this line too

**Comment by neutrinoceros** [comment]
Can we merge this to ease up work on the last compilation error ?
@chrishavlin @matthewturk ?

**Review by chrishavlin — APPROVED** [review]

**Comment by chrishavlin** [comment]
> Can we merge this to ease up work on the last compilation error ?

OK by me -- I assume bumping the minimum version of cython from 0.29.21 to 0.29.33 shouldn't be an issue, right? seems totally reasonable to me... 

@matthewturk ?

**Pull request merged**